ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PointSchedule/Shrinking.hs

```diff
@@ -60,6 +73,68 @@ shrinkOtherPeers shrink Peers{honest, others} =
   map (Peers honest . Map.fromList) $
     shrinkList (traverse (traverse shrink)) $ Map.toList others
 
+-- | Shrinks an honest peer by removing ticks.
+-- Because we are manipulating `PeerSchedule` at that point, there is no proper
+-- notion of a tick. Instead, we remove points of the honest `PeerSchedule`,
+-- and move all other points sooner, including those on the adversarial schedule.
+-- We check that this operation neither changes the final state of the honest peer,
+-- nor that it removes points from the adversarial schedules.
+shrinkHonestPeer :: Peers (PeerSchedule blk) -> [Peers (PeerSchedule blk)]
+shrinkHonestPeer Peers{honest, others} = do
+  (at, speedUpBy) <- splits
+  (honest', others') <- maybeToList $ do
+    honest' <- traverse (speedUpHonestSchedule at speedUpBy) honest
+    others' <- mapM (traverse (speedUpAdversarialSchedule at speedUpBy)) others
+    pure (honest', others')
+  pure $ Peers honest' others'
+  where
+    -- | A list of non-zero time intervals between successive points of the honest schedule
+    splits :: [(Time, DiffTime)]
+    splits = mapMaybe
+      (\((t1, _), (t2, _)) ->
+        if t1 == t2
+          then Nothing
+          else Just (t1, diffTime t2 t1)
+      )
+      (zip (value honest) (drop 1 $ value honest))
+
+-- | Speeds up an honest schedule after `at` time, by `speedUpBy`.
+-- This "speeding up" is done by removing `speedUpBy` to all points after `at`,
+-- and removing those points if they fall before `at`. We check that the operation
+-- doesn't change the final state of the peer, i.e. it doesn't remove all TP, HP, and BP
+-- in the sped up part.
+speedUpHonestSchedule :: Time -> DiffTime -> PeerSchedule blk -> Maybe (PeerSchedule blk)
+speedUpHonestSchedule at speedUpBy sch =
```

Comment:
From what I can see this function doesn't change the "speed" but rather shifts time. When I was reading this I was expecting to find a transformation by means of multiplying rather than subtraction. Consider renaming this function.

Reply:
I agree that the name is poorly chosen. The function in question has completely disappeared from milestone 13, so I'd be up for keeping it with the current name in the meantime. I am happy to rename it to `shiftHonestSchedule` (and same for the adversarial one) if you prefer this, though!